pkg/lib/account_identifier.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package lib
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/moov-io/bai2/pkg/util"
+)
+
+const (
+	aiParseErrorFmt    = "AccountIdentifier: unable to parse %s"
+	aiValidateErrorFmt = "AccountIdentifierCurrent: invalid %s"
+)
+
+/*
+
+Account Identifier and Summary/Status for Current (CDA), Personal (PDA), and Loan accounts
+
+CDA and PDA accounts
+This record contains information on opening and closing balances for CDA and PDA accounts. It always
+has a record code of 03.
+
+*/
+
+// Creating CDA and PDA accounts
+func NewAccountIdentifier() *AccountIdentifier {
+	return &AccountIdentifier{
+		RecordCode: "03",
+	}
+}
+
+// CDA and PDA accounts
+type AccountIdentifier struct {
+	RecordCode    string
+	AccountNumber string
+	CurrencyCode  string   `json:",omitempty"`
+	TypeCode      string   `json:",omitempty"`
+	Amount        string   `json:",omitempty"`
```

Comment:
Can we make fields that are semantically numbers into ints? I don't think it's necessary for record and type codes, but this would be really useful for amounts, counts and totals.

Reply:
In fact, we need to add new type (signed number)
signed number allow sign symbol, comma, point, and ext symbol. 
of course, we need to implement operation override feature for new signed number too

So used string type instead of the type

Reply:
I'm not sure I follow. Are you saying we need to add a new type for `Amount`s so it can support different decimal precision by currency type? If so, I agree that would be good, but I don't understand what you mean by "allow sign symbol, comma, point, and ext symbol", and I certainly don't think it warrants a whole new _numeric_ type (and Go doesn't support operator overloading). It's definitely out of scope for this PR though, so for right now it's probably best to leave it as a string and let the user sort it out.

Things like `RecordCount` can be safely made into integers, though.

Reply:
updated ItemCount with int type